api/snyk.ts

```diff
@@ -0,0 +1,49 @@
+import cheerio from 'cheerio'
+import got from '../libs/got'
+import { createBadgenHandler, PathArgs, Query } from '../libs/create-badgen-handler'
+
+export default createBadgenHandler({
+  title: 'Snyk',
+  examples: {
+    '/snyk/badgen/badgen.net': 'vulnerability scan',
+    '/snyk/babel/babel/6.x': 'vulnerability scan (branch)',
+    '/snyk/tunnckoCore/opensource?targetFile=@tunnckocore/utils/package.json': 'vulnerability scan (custom path)'
```

Comment:
We need a better schema for this, to keep the convention of "path args for content, query args for style"

Reply:
How about converting the `targetFile` to matrix param? 🤔 https://www.logicbig.com/quick-info/web/matrix-param.html

Reply:
Great idea! but still feels a bit tricky to most people didn't know about matrix param before

Reply:
Maybe we need to choose between `branch` and `targetFile` then 😂

Which one would be most used? We can add the other one later as a nicer solution comes out.

Reply:
> Maybe we need to choose between `branch` and `targetFile` then 😂
> 
> Which one would be most used? We can add the other one later as a nicer solution comes out.

That is a really hard question. It is quite often that projects have a dev branch set as the main one and periodically push stable code dumps to master/release. Now it is a question whether they want to use the badge as dev indicator on in-flux dev branch or as a message to end-users as part of their stable release. Maybe they want both making branch altering quite necessary. OTOH if they have monorepo they must have the ability to target partial packages by pointing to appropriate manifest/s. I can't really say which would be more useful. But if one must go it probably makes sense to let go `targetFile` because as an existing badge user I wouldn't expect that to be implemented 🙃

Reply:
Or we can support `targetFile` in additional path:
- `/snyk/:user/:repo/:branch?`
- `/snyk/:user/:repo/:branch/:targetFile+`

to monorepo user (needs `targetFile`), still better than nothing. to none-monorepo user (the majority), it's the best.

Reply:
It is a nice compromise but there is a nasty edge case when git branch contains slash. In that case, you can't really safely assume where branch name ends and pathname starts. 🙄

Reply:
🤦‍♂️

Reply:
Technically, we can use `%2F` as slash in branch section

Reply:
Yeah, that would work! Let's go with that then. I'll update PR accordingly.